### PR TITLE
hack: improve git tag pushing

### PR DESCRIPTION
**Pull request by jpeach**
- Do a dry-run to test that the release tag can be pushed
- Fix the instructional output to show hot to push just the new tag
- Check there there are YAML changes before trying to commit

Signed-off-by: James Peach <jpeach@vmware.com>

### CODE REVIEW

File: hack/release/make-release-tag.sh [hunk 1]
```diff
@@ -57,4 +61,8 @@ $(git shortlog "$OLDVERS..HEAD")
 EOF
 
 printf "Created tag '%s'\n" "$NEWVERS"
-printf "Run '%s' to push the tag if you are happy\n" "git push --tags"
+
+printf "Testing whether tag '%s' can be pushed\n" "$NEWVERS"
+git push --dry-run origin "$NEWVERS"
```
Comment (stevesloka) [inline hack/release/make-release-tag.sh]: For me `origin` is my personal fork. I know for you @jpeach it's the upstream projectcontour repo. Not sure how to solve that in this script.

**Comment by jpeach** [comment]
On Mar 28, 2020, at 4:30 AM, Steve Sloka <notifications@github.com> wrote:

For me origin is my personal fork. I know for you @jpeach<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjpeach&data=02%7C01%7Cjpeach%40vmware.com%7C878eb93c0f4e4ea6df2a08d7d2748a2d%7Cb39138ca3cee4b4aa4d6cd83d9dd62f0%7C0%7C0%7C637209270286398453&sdata=HXOMhsazeUgo%2FkFXJaI751jgkN%2BvZ7Z8e2DtP0P%2FQoI%3D&reserved=0> it's the upstream projectcontour repo. Not sure how to solve that in this script.

That’s a good point. I wanted to drop the advice to push all tags, which is a bit dangerous. I’ll see whether I can figure out the right remote automatically.

File: hack/release/make-release-tag.sh (hunk 1 above)
Comment (jpeach) [reply hack/release/make-release-tag.sh]: I added some code to fish out a likely-looking remote. Does that work correctly for your config?

Comment (youngnick) [reply hack/release/make-release-tag.sh]: That looks like a reasonable guess to me.

**Comment by jpeach** [comment]
@stevesloka Can you please check that the remote detection change is correct for your workflow?

**Pull request merged**